whois-api/src/main/java/net/ripe/db/whois/api/rdap/RdapExceptionMapper.java

```diff
@@ -46,6 +47,18 @@ public Response toResponse(final Exception exception) {
             return Response.status(HttpServletResponse.SC_BAD_REQUEST).entity(createErrorEntity(HttpServletResponse.SC_BAD_REQUEST, exception.getMessage())).build();
         }
 
+        if (exception instanceof ParamException && ((ParamException) exception).getParameterName().equals("objectType")) {
+            return Response.status(HttpServletResponse.SC_BAD_REQUEST).entity(createErrorEntity(HttpServletResponse.SC_BAD_REQUEST, "unknown type")).build();
+        }
+
+        if (exception instanceof NotFoundException) {
```

Comment:
Are NotFound and BadRequest not already covered by the WebApplicationException catch-all immediately below?

Reply:
As discussed, wrap Bad Request and Not Found in an RDAP JSON response. The WebApplicationException is generic, and should never happen, so log a warning and return the plain text error in that case.